### DISCUSSION TITLE
How to render text in world space?

### DISCUSSION THREAD

**Discussion by fishmans97** [vsg-dev/VulkanSceneGraph#1233] [Q&A] [answered]
does the vsg::text only available in screen space?

**Comment by robertosfield** [comment, marked as answer]
vsg::Text is rendered in whatever coordinate frame you place it in, but that screen space or world space.  You just add the vsg::Text node to your scene graph like any other object you want rendered.

Almost all the examples put the scene into world coordinates and this includes the two text related examples [vsgtext](https://github.com/vsg-dev/vsgExamples/blob/master/examples/text/vsgtext/vsgtext.cpp) and [vsgtextgroup](https://github.com/vsg-dev/vsgExamples/blob/master/examples/text/vsgtextgroup/vsgtextgroup.cpp).

I'm a bit perplexed why you might think that vsg::Text is only available in screen space.  As far as I'm aware there is nothing in documentation, discussions on the forum, or the examples that would suggest this. 

Could it be you asking the more advanced question about rendering text in ECEF (Earth Center Earth Fixed) as required by GIS and whole earth/space applications?  Just place the text under a vsg::MatrxTrasform to place text set up in local origin into ECEF.

**Reply by fishmans97** [reply]
Thanks for your answer! I set the position of the text in the ECEF coordinate system, and it appears there. However, when I move my camera, the position of the text changes, and it only appears in the upper left corner of the screen. Beyond this area, it feels like it has been cropped out. I checked the settings of the viewport and found no issues.  In the end, I discovered the cause of the problem, which was that all of my sub scenes had their VkSampleCountFlagBits values set to 8, except for the text I added.
Then,i add the code that "shaderSet->defaultGraphicsPipelineStates.push_back(veg::MultisampleState::create (VK_SAMPLE_COUNT_8_BIT))",it works correctly!
So, i apologize for disturbing your work as it is a personal code issue, and I greatly appreciate your answer!
